rasterio/windows.py

```diff
@@ -88,5 +140,172 @@ def intersect(*windows):
     boolean:
         True if all windows intersect.
     """
-    from rasterio._io import windows_intersect
-    return windows_intersect(windows)
+
+    from itertools import combinations
+
+    def intersects(range1, range2):
+        return not (
+            range1[0] >= range2[1] or range1[1] <= range2[0]
+        )
+
+    windows = np.array(windows)
+
+    for i in (0, 1):
+        for c in combinations(windows[:, i], 2):
+            if not intersects(*c):
+                return False
+
+    return True
+
+
+def window(transform, left, bottom, right, top,
+           height=None, width=None, boundless=False):
+    """Returns the window corresponding to the world bounding box.
+    If boundless is False, window is limited to extent of the
+    data (determined by transform, height and width)."""
+
+    window = get_window(left, bottom, right, top, transform)
+    if boundless:
+        return window
+    else:
+        if None in (height, width):
+            raise ValueError("Must supply height and width unless boundless")
+        return crop_window(window, height, width)
+
+
+def window_transform(transform, window):
+    """Returns the affine transform for a dataset window."""
+    (r, _), (c, _) = window
+    return transform * Affine.translation(c or 0, r or 0)
+
+
+def window_bounds(transform, window):
+    """Returns the bounds of a window as x_min, y_min, x_max, y_max."""
+    ((row_min, row_max), (col_min, col_max)) = window
+    x_min, y_min = transform * (col_min, row_max)
+    x_max, y_max = transform * (col_max, row_min)
+    return x_min, y_min, x_max, y_max
+
+
+def crop_window(window, height, width):
```

Comment:
@perrygeo now that this and other functions above and below are namespaced in `rasterio.windows` consider dropping `window_*` and `*_window` from the function names.